src/NServiceBus.Core/Transports/Msmq/LimitedConcurrencyLevelTaskScheduler.cs

```diff
@@ -0,0 +1,129 @@
+namespace NServiceBus.Transports.Msmq
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+
+    //for now
+    //copy and pasted from https://msdn.microsoft.com/en-us/library/ee789351(v=vs.110).aspx
+    class LimitedConcurrencyLevelTaskScheduler : TaskScheduler
```

Comment:
Not sure, do we?  (ms doesn't specify a license on the page, its just an "example"?)